bpf-gpl/skb.h

```diff
@@ -118,12 +123,17 @@ static CALI_BPF_INLINE void skb_refresh_hdr_ptrs(struct cali_tc_ctx *ctx)
  * - ctx->data_start/end
  * - ctx->eth (if this BPF program has access to the L2 header)
  * - ctx->ip_header
- * - ctx->nh/tcp_header/udp_header/icp_header.
+ * - ctx->nh/tcp_header/udp_header/icmp_header.
  */
 static CALI_BPF_INLINE int skb_refresh_validate_ptrs(struct cali_tc_ctx *ctx, long nh_len) {
 	int min_size = skb_iphdr_offset(ctx->skb) + IP_SIZE;
 	skb_refresh_start_end(ctx);
 	if (ctx->data_start + (min_size + nh_len) > ctx->data_end) {
+		// This is an XDP program and there is not enough data for next header.
+		if (CALI_F_XDP) {
```

Comment:
Would it be useful to add a CALI_DEBUG here?  Can we get tracing for XDP programs?

Reply:
We do. Added a CALI_DEBUG here.